### PR TITLE
Development3

### PR DESCRIPTION
**Pull request by devsincmateen**
Pull request for the upcoming code

### CODE REVIEW

File: app/controllers/features_controller.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+class FeaturesController < ApplicationController
+  def index
+    @features = Feature.all
+  end
+
+  def create
+    @feature = Feature.new(feature_params)
+    p @features
```
Comment (waqarnazir) [inline app/controllers/features_controller.rb:10]: what is this line for??

File: app/controllers/features_controller.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  def new
+    @feature = Feature.new
+  end
+
+  def edit
+    @feature = Feature.find(params[:id])
+  end
+
+  def show
+    @feature = Feature.find(params[:id])
```
Comment (waqarnazir) [inline app/controllers/features_controller.rb:27]: this logic is common, move it to a private method and use callbacks

File: app/controllers/features_controller.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  end
+
+  def update
+    @feature = Feature.find(params[:id])
```
Comment (waqarnazir) [inline app/controllers/features_controller.rb:31]: same as above

File: app/controllers/features_controller.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    if @feature.update(feature_params)
+      redirect_to @feature
+    else
+      redirect_to 'edit'
+    end
+  end
+
+  def destroy; end
```
Comment (waqarnazir) [inline app/controllers/features_controller.rb:39]: why nothing in destroy??

File: app/controllers/plans_controller.rb [hunk 5]
```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+class PlansController < ApplicationController
+  def index
+    @plans = Plan.all
+  end
+
+  def create
+    @plan = Plan.new(plan_params)
+    p @plans
```
Comment (waqarnazir) [inline app/controllers/plans_controller.rb:10]: what is this doing?

File: app/controllers/plans_controller.rb [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  def edit
+    @plan = Plan.find(params[:id])
+  end
+
+  def show
+    @plan = Plan.find(params[:id])
+  end
+
+  def update
+    @plan = Plan.find(params[:id])
```
Comment (waqarnazir) [inline app/controllers/plans_controller.rb:32]: same

File: app/controllers/users_controller.rb [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+
+class UsersController < ApplicationController
+  def index
+    @users = User.all
+  end
+
+  def delete
+    @user = User.find(params[:id])
+    @user.destroy
+  end
```
Comment (waqarnazir) [inline app/controllers/users_controller.rb:11]: why users are being manipulated ??

File: app/models/plan.rb [hunk 8]
```diff
@@ -0,0 +1,5 @@
+# frozen_string_literal: true
+
+class Plan < ApplicationRecord
+    has_many :Features
```
Comment (waqarnazir) [inline app/models/plan.rb:4]: why is it capital, should be :features. where are dependent options.

File: app/models/user.rb [hunk 9]
```diff
@@ -1,7 +1,23 @@
+# frozen_string_literal: true
+
+# Users class for devise
 class User < ApplicationRecord
+  # validates :email, format: { with: URI::MailTo::EMAIL_REGEXP }, presence: true
+  # validates :password, presence: true,
+  #                      confirmation: true,
+  #                      length: { within: 6..40 },
+  #                      unless: :force_submit
+
```
Comment (waqarnazir) [inline app/models/user.rb:10]: why validations are commented?

File: app/views/plans/_plan.html.erb [hunk 10]
```diff
@@ -0,0 +1,6 @@
+<tr>
+      <td><%= plan.name %></td>
+      <td><%= plan.code %></td>
+      <td><%= plan.monthly_income %></td>
+     
+</tr>
```
Comment (waqarnazir) [inline app/views/plans/_plan.html.erb:6]: indentation should be single tabbed

**Review by waqarnazir — CHANGES_REQUESTED** [review]

**Comment by waqarnazir** [comment]
Remove unnecessary empty files, Make sure rubocop is green. Check rails basic guidelines for references where stuck. Remove commented code.

File: app/controllers/features_controller.rb (hunk 1 above)
Comment (devsincmateen) [reply app/controllers/features_controller.rb:10]: just for debuging removed

File: app/controllers/features_controller.rb (hunk 4 above)
Comment (devsincmateen) [reply app/controllers/features_controller.rb:39]: under construction

File: app/controllers/users_controller.rb (hunk 7 above)
Comment (devsincmateen) [reply app/controllers/users_controller.rb:11]: because admin has to manage them

File: app/models/plan.rb (hunk 8 above)
Comment (devsincmateen) [reply app/models/plan.rb:4]: updated

File: app/models/user.rb (hunk 9 above)
Comment (devsincmateen) [reply app/models/user.rb:10]: temporary

**Pull request merged**